<a href="https://colab.research.google.com/github/shaloy-lewis/EDA_on_play_store_app_reviews/blob/main/EDA_on_Play_Store_App_Reviews_by_Shaloy_Lewis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

* We are provided with two sets of .csv files which contains the datasets required to analyse the playstore app reviews. By analysing these two datasets, we must come up with the key factors that are responsible for app engagement and success.
* Let us start by analysing the 'play store data.csv' file.

In [ ]:
# Importing the required libraries to read the csv files

import pandas as pd
import numpy as np

# **Exploring the playstoredata.csv file**

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
# Reading the csv file

play_store_data_path = '/content/drive/MyDrive/EDA on Play Store App Reviews - Shaloy Lewis/Play Store Data.csv'

ps_df = pd.read_csv(play_store_data_path)

In [ ]:
# Basic inspection
# Viewing the top 10 rows of the imported dataframe

ps_df.head(10)

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up
5                         Paper flowers instructions  ...    2.3 and up
6            Smoke Effect Photo Maker - Smoke Editor  ...  4.0.3 and up
7                                   Infinite Painter  ...    4.2 and up
8                               Garden Coloring Book  ...    3.0 and up
9                      Kids Paint Free - Drawing Fun  ...  4.0.3 and up

[10 rows x 13 columns]

In [ ]:
# Viewing the bottom 10 rows of the dataframe

ps_df.tail(10)

App  ...         Android Ver
10831                       payermonstationnement.fr  ...          4.0 and up
10832                                       FR Tides  ...          2.1 and up
10833                                    Chemin (fr)  ...          2.2 and up
10834                                  FR Calculator  ...          4.1 and up
10835                                       FR Forms  ...          4.0 and up
10836                               Sya9a Maroc - FR  ...          4.1 and up
10837               Fr. Mike Schmitz Audio Teachings  ...          4.1 and up
10838                         Parkinson Exercices FR  ...          2.2 and up
10839                  The SCP Foundation DB fr nn5n  ...  Varies with device
10840  iHoroscope - 2018 Daily Horoscope & Astrology  ...  Varies with device

[10 rows x 13 columns]

In [ ]:
# Finding the number of rows and columns in the given dataset

ps_df.shape

(10841, 13)

Hence, we can say that there are a total of 10841 rows and 13 columns in this .csv file. Let us explore the columns, and the significance of the data it possesses.

In [ ]:
# Columns of this dataset are

ps_df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

Let us first define what information the columns contain based on our inspection.

* App: Contains the name of the app with a short description (optional).
* Category: It gives the category to the app.
* Rating: It contains the average rating the respective app received from its users.
* Reviews: It contains the number of users that have dropped a review for the respective app.
* Size: It contains the the disk space required to install the respective app.
* Installs: It gives the rounded figure of number of times the respective app was downloaded.
* Type: It states whether an app is free to use or paid.
* Price: It gives the price payable to install the app. For free type apps, the price is zero.
* Content rating: It states whether or not an app is suitable for all age groups or not.
* Genres: It gives the genre(s) to which the respective app belongs.
* Last updated: It gives the day in which the latest update was released.
* Current ver: It gives the current version of the respective app.
* Android ver: It gives the android version of the respective app.

# **Handling the errors and NaN values one column at a time.**

In [ ]:
# First of all lets find the number of NaN values in each column

ps_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

Let us try to handle these NaN values one column at a time.

Lets find the percentage of of rows that contain atleast one NaN value.

In [ ]:
# Percentage of rows containing atleast one NaN value

total_nan = sum([True for id,row in ps_df.iterrows() if any(row.isnull())])

perc_nan = round(total_nan/10841*100, 2)

print(f'Hence, there are a total of {total_nan}({perc_nan}%) rows which contain atleast one NaN value.')

Hence, there are a total of 1481(13.66%) rows which contain atleast one NaN value.


Since approximately 13.66% of all the rows in the dataset contains atleast one NaN value, it is not practcal to eliminate the rows containing NaN values. We need to handle this problem in another way.

Lets first deal with the columns which contains lesser number of NaN values. By going through the NaN values, we must come up with a way to replace them with non NaN values or we need to come up with a reason for having NaN.
1. Android Ver:
There are a total of 3 NaN values in this column.

In [ ]:
# The rows containing NaN values in the Android Ver column

ps_df[ps_df['Android Ver'].isnull()]

App  ... Android Ver
4453                    [substratum] Vacuum: P  ...         NaN
4490                      Pi Dark [substratum]  ...         NaN
10472  Life Made WI-Fi Touchscreen Photo Frame  ...         NaN

[3 rows x 13 columns]

* There are total of three entries with NaN in the 'Android Ver' column.
* There are a lot of errors the row corresponding to index number 10472. For example, the category contains the value '1.9', which does not make any sense, the rating is '19' which cannot be true, since the play store app reviews can be in between 1 and 5 only, and so on. So it is of our best interest to delete this row altogather so as to eliminate any errors as we proceed with the analysis.

In [ ]:
# Dropping the row corresponding to index number 10472

ps_df.drop(10472, inplace= True)

In [ ]:
# Shape of the updated dataframe

ps_df.shape

(10840, 13)

In [ ]:
# Finding the different values the 'Android Ver' column takes

ps_df['Android Ver'].value_counts()

4.1 and up            2451
4.0.3 and up          1501
4.0 and up            1375
Varies with device    1362
4.4 and up             980
2.3 and up             652
5.0 and up             601
4.2 and up             394
2.3.3 and up           281
2.2 and up             244
4.3 and up             243
3.0 and up             241
2.1 and up             134
1.6 and up             116
6.0 and up              60
7.0 and up              42
3.2 and up              36
2.0 and up              32
5.1 and up              24
1.5 and up              20
4.4W and up             12
3.1 and up              10
2.0.1 and up             7
8.0 and up               6
7.1 and up               3
5.0 - 8.0                2
4.0.3 - 7.1.1            2
1.0 and up               2
2.2 - 7.1.1              1
4.1 - 7.1.1              1
5.0 - 7.1.1              1
7.0 - 7.1.1              1
5.0 - 6.0                1
Name: Android Ver, dtype: int64

Since the remaining NaN values in the 'Android Ver' column cannot be replaced by any particular value, it is left as it is.

In [ ]:
# NaN values in each column

ps_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       0
Genres               0
Last Updated         0
Current Ver          8
Android Ver          2
dtype: int64

2. Current Ver: There are a total of 8 NaN values in this column.

In [ ]:
# The rows containing NaN values in the Current Ver column

ps_df[ps_df['Current Ver'].isnull()]

App  ... Android Ver
15     Learn To Draw Kawaii Characters  ...  4.2 and up
1553              Market Update Helper  ...  1.5 and up
6322            Virtual DJ Sound Mixer  ...  4.0 and up
6803                         BT Master  ...  1.6 and up
7333                       Dots puzzle  ...  4.0 and up
7407                   Calculate My IQ  ...  2.3 and up
7730                            UFO-CQ  ...  2.0 and up
10342                   La Fe de Jesus  ...  3.0 and up

[8 rows x 13 columns]

In [ ]:
# Finding the different values the 'Current Ver' column takes

ps_df['Current Ver'].value_counts()

Varies with device    1459
1.0                    809
1.1                    264
1.2                    178
2.0                    151
                      ... 
2.0.833.18               1
0.33                     1
2.5.1.7592               1
1.131                    1
1.0.612933               1
Name: Current Ver, Length: 2831, dtype: int64

Since the NaN values in the 'Current Ver' column cannot be replaced by any particular value, it is left as it is.

3. Type: There is only one NaN value in this column. Lets see whether or not we can handle this value.

In [ ]:
# The rows containing NaN values in the Current Ver column

ps_df[ps_df['Type'].isnull()]

App  ...         Android Ver
9148  Command & Conquer: Rivals  ...  Varies with device

[1 rows x 13 columns]

In [ ]:
# Finding the different values the 'Type' column takes

ps_df['Type'].value_counts()

Free    10039
Paid      800
Name: Type, dtype: int64

The 'Type' column contains only two entries, namely, 'Free' and 'Paid'. Also, if the app is of type-paid, the price of that app will be printed in the corresponding 'Price' column, else, it will show as '0'.In this case, the price for the respective app is printed as '0', which means the app is of type-free. Hence we can replace this NaN value with 'Free'.

In [ ]:
# Replacing the NaN value in 'Type' column corresponding to row index 9148 with 'Free'

ps_df.loc[9148,'Type'] = 'Free'

In [ ]:
# Finding the different values the 'Type' column takes

ps_df['Type'].value_counts()

Free    10040
Paid      800
Name: Type, dtype: int64

In [ ]:
# NaN values in each column

ps_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 0
Price                0
Content Rating       0
Genres               0
Last Updated         0
Current Ver          8
Android Ver          2
dtype: int64

4. Rating: This column contains 1474 NaN values.

In [ ]:
# The rows containing NaN values in the Rating column

ps_df[ps_df['Rating'].isnull()]

App  ... Android Ver
23                  Mcqueen Coloring pages  ...  4.1 and up
113              Wrinkles and rejuvenation  ...  3.0 and up
123                 Manicure - nail design  ...  4.1 and up
126           Skin Care and Natural Beauty  ...  4.1 and up
129    Secrets of beauty, youth and health  ...  2.3 and up
...                                    ...  ...         ...
10824                            Cardio-FR  ...  4.4 and up
10825                   Naruto & Boruto FR  ...  4.0 and up
10831             payermonstationnement.fr  ...  4.0 and up
10835                             FR Forms  ...  4.0 and up
10838               Parkinson Exercices FR  ...  2.2 and up

[1474 rows x 13 columns]

In [ ]:
# Finding mean and median in the Rating column excluding the NaN values.

mean_rating = round(ps_df[~ps_df['Rating'].isnull()]['Rating'].mean(),1)

median_rating = ps_df[~ps_df['Rating'].isnull()]['Rating'].median()

[mean_rating, median_rating]

[4.2, 4.3]

In [ ]:
# Replacing the NaN values in the 'Rating' colunm with its median value

ps_df['Rating'].fillna(value=median_rating, inplace=True)

In [ ]:
# NaN values in each column

ps_df.isnull().sum()

App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              0
Price             0
Content Rating    0
Genres            0
Last Updated      0
Current Ver       8
Android Ver       2
dtype: int64

In [ ]:
# Cross checking the value of median in the 'Rating' column

ps_df['Rating'].median()

4.3

Now that we have taken care of the NaN values in all columns except for a few exceptions in 'Current Ver' and 'Android Ver' columns, Let us make some slight changes in the data so that the data can be analysed easier.

In [ ]:
ps_df.head()

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up

[5 rows x 13 columns]

In [ ]:
ps_df['Last Updated'].value_counts()

August 3, 2018     326
August 2, 2018     304
July 31, 2018      294
August 1, 2018     285
July 30, 2018      211
                  ... 
August 28, 2016      1
January 9, 2015      1
May 2, 2016          1
August 9, 2013       1
June 1, 2012         1
Name: Last Updated, Length: 1377, dtype: int64

In [ ]:
ps_df['Genres'].value_counts()

Tools                                  842
Entertainment                          623
Education                              549
Medical                                463
Business                               460
                                      ... 
Strategy;Creativity                      1
Health & Fitness;Action & Adventure      1
Tools;Education                          1
Music & Audio;Music & Video              1
Puzzle;Education                         1
Name: Genres, Length: 119, dtype: int64

In [ ]:
ps_df['Content Rating'].value_counts()

Everyone           8714
Teen               1208
Mature 17+          499
Everyone 10+        414
Adults only 18+       3
Unrated               2
Name: Content Rating, dtype: int64

In [ ]:
ps_df['Price'].value_counts()

0          10040
$0.99        148
$2.99        129
$1.99         73
$4.99         72
           ...  
$109.99        1
$1.50          1
$37.99         1
$3.61          1
$28.99         1
Name: Price, Length: 92, dtype: int64

In [ ]:
ps_df['Type'].value_counts()

Free    10040
Paid      800
Name: Type, dtype: int64

In [ ]:
ps_df['Installs'].value_counts()

1,000,000+        1579
10,000,000+       1252
100,000+          1169
10,000+           1054
1,000+             907
5,000,000+         752
100+               719
500,000+           539
50,000+            479
5,000+             477
100,000,000+       409
10+                386
500+               330
50,000,000+        289
50+                205
5+                  82
500,000,000+        72
1+                  67
1,000,000,000+      58
0+                  14
0                    1
Name: Installs, dtype: int64

In [ ]:
ps_df['Size'].value_counts()

Varies with device    1695
11M                    198
12M                    196
14M                    194
13M                    191
                      ... 
496k                     1
353k                     1
545k                     1
499k                     1
308k                     1
Name: Size, Length: 461, dtype: int64

In [ ]:
ps_df['Reviews'].value_counts()

0        596
1        272
2        214
3        175
4        137
        ... 
3250       1
1853       1
4228       1
21147      1
13166      1
Name: Reviews, Length: 6001, dtype: int64

In [ ]:
ps_df['Rating'].value_counts()

4.3    2550
4.4    1109
4.5    1038
4.2     952
4.6     823
4.1     708
4.0     568
4.7     499
3.9     386
3.8     303
5.0     274
3.7     239
4.8     234
3.6     174
3.5     163
3.4     128
3.3     102
4.9      87
3.0      83
3.1      69
3.2      64
2.9      45
2.8      42
2.6      25
2.7      25
2.5      21
2.3      20
2.4      19
1.0      16
2.2      14
1.9      13
2.0      12
1.7       8
2.1       8
1.8       8
1.6       4
1.4       3
1.5       3
1.2       1
Name: Rating, dtype: int64

In [ ]:
ps_df['Category'].value_counts()

FAMILY                 1972
GAME                   1144
TOOLS                   843
MEDICAL                 463
BUSINESS                460
PRODUCTIVITY            424
PERSONALIZATION         392
COMMUNICATION           387
SPORTS                  384
LIFESTYLE               382
FINANCE                 366
HEALTH_AND_FITNESS      341
PHOTOGRAPHY             335
SOCIAL                  295
NEWS_AND_MAGAZINES      283
SHOPPING                260
TRAVEL_AND_LOCAL        258
DATING                  234
BOOKS_AND_REFERENCE     231
VIDEO_PLAYERS           175
EDUCATION               156
ENTERTAINMENT           149
MAPS_AND_NAVIGATION     137
FOOD_AND_DRINK          127
HOUSE_AND_HOME           88
AUTO_AND_VEHICLES        85
LIBRARIES_AND_DEMO       85
WEATHER                  82
ART_AND_DESIGN           65
EVENTS                   64
PARENTING                60
COMICS                   60
BEAUTY                   53
Name: Category, dtype: int64

In [ ]:
ps_df['App'].value_counts()

ROBLOX                                               9
CBS Sports App - Scores, News, Stats & Watch Live    8
Candy Crush Saga                                     7
8 Ball Pool                                          7
Duolingo: Learn Languages Free                       7
                                                    ..
DREAM-e: Dream Analysis A.I.                         1
English-Myanmar Dictionary                           1
Horses Live Wallpaper                                1
FL Tax-Verify                                        1
NeuroNation - Focus and Brain Training               1
Name: App, Length: 9659, dtype: int64

In [ ]:
ps_df[ps_df['App'] == 'ROBLOX']

App Category  Rating  ... Last Updated   Current Ver Android Ver
1653  ROBLOX     GAME     4.5  ...   2018-07-31  2.347.225742  4.1 and up
1701  ROBLOX     GAME     4.5  ...   2018-07-31  2.347.225742  4.1 and up
1748  ROBLOX     GAME     4.5  ...   2018-07-31  2.347.225742  4.1 and up
1841  ROBLOX     GAME     4.5  ...   2018-07-31  2.347.225742  4.1 and up
1870  ROBLOX     GAME     4.5  ...   2018-07-31  2.347.225742  4.1 and up
2016  ROBLOX   FAMILY     4.5  ...   2018-07-31  2.347.225742  4.1 and up
2088  ROBLOX   FAMILY     4.5  ...   2018-07-31  2.347.225742  4.1 and up
2206  ROBLOX   FAMILY     4.5  ...   2018-07-31  2.347.225742  4.1 and up
4527  ROBLOX   FAMILY     4.5  ...   2018-07-31  2.347.225742  4.1 and up

[9 rows x 13 columns]

In [ ]:
drop_duplicates

In [ ]:
ps_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10840 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10840 non-null  object 
 1   Category        10840 non-null  object 
 2   Rating          10840 non-null  float64
 3   Reviews         10840 non-null  object 
 4   Size            10840 non-null  object 
 5   Installs        10840 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10840 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10840 non-null  object 
 10  Last Updated    10840 non-null  object 
 11  Current Ver     10832 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.5+ MB


In [ ]:
ps_df.describe()

Rating
count  10840.000000
mean       4.206476
std        0.480342
min        1.000000
25%        4.100000
50%        4.300000
75%        4.500000
max        5.000000

* Only the 'Rating' column is of float datatype. We need to convert the 'Reviews', 'Size', 'Installs' and 'Price' columns into float/int datatype so that we can perform quantitative analysis on it.
* It will be easy to perform the analysis on the 'Size' column if all the values are of the same unit, ie, either in MB or KB.
* We need to convert the 'Last updated' column from str to datetime datatype to perform chronological analysis.

Converting the values in the 'Last Updated' column from str to datetime datatype.

In [ ]:
# importing the datetime library

from datetime import datetime

In [ ]:
new_df = ps_df['Last Updated'].apply(lambda x: datetime.strptime(x,'%B %d, %Y'))
new_df

0       2018-01-07
1       2018-01-15
2       2018-08-01
3       2018-06-08
4       2018-06-20
           ...    
10836   2017-07-25
10837   2018-07-06
10838   2017-01-20
10839   2015-01-19
10840   2018-07-25
Name: Last Updated, Length: 10840, dtype: datetime64[ns]

In [ ]:
ps_df['Last Updated'] = ps_df['Last Updated'].apply(lambda x: datetime.strptime(x,'%B %d, %Y'))
ps_df.head()

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up

[5 rows x 13 columns]

In [ ]:
ps_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10840 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   App             10840 non-null  object        
 1   Category        10840 non-null  object        
 2   Rating          10840 non-null  float64       
 3   Reviews         10840 non-null  object        
 4   Size            10840 non-null  object        
 5   Installs        10840 non-null  object        
 6   Type            10840 non-null  object        
 7   Price           10840 non-null  object        
 8   Content Rating  10840 non-null  object        
 9   Genres          10840 non-null  object        
 10  Last Updated    10840 non-null  datetime64[ns]
 11  Current Ver     10832 non-null  object        
 12  Android Ver     10838 non-null  object        
dtypes: datetime64[ns](1), float64(1), object(11)
memory usage: 1.5+ MB


In [ ]:
test = ['$499', '$2', '$67', '0', '$0' ,'55', '$2.365']

In [ ]:
def drop_dollar(val):
  new_list = []
  newer_list = []
  for i in val:
    if '$' in i:
      new_list.append(i[1:])
    else:
      new_list.append(i)
  # return new_list
  for j in new_list:
    newer_list.append(float(j))
  return newer_list    

  # return val[:][1:]

In [ ]:
drop_dollar(test)

[499.0, 2.0, 67.0, 0.0, 0.0, 55.0, 2.365]

# **Exploring userreviews.csv file**

In [ ]:
user_reviews_data_path = '/content/drive/MyDrive/EDA on Play Store App Reviews - Shaloy Lewis/User Reviews.csv'

ur_df = pd.read_csv(user_reviews_data_path)

In [ ]:
ur_df.head(10)

App  ... Sentiment_Subjectivity
0  10 Best Foods for You  ...               0.533333
1  10 Best Foods for You  ...               0.288462
2  10 Best Foods for You  ...                    NaN
3  10 Best Foods for You  ...               0.875000
4  10 Best Foods for You  ...               0.300000
5  10 Best Foods for You  ...               0.300000
6  10 Best Foods for You  ...               0.900000
7  10 Best Foods for You  ...                    NaN
8  10 Best Foods for You  ...               0.000000
9  10 Best Foods for You  ...               0.000000

[10 rows x 5 columns]

In [ ]:
ur_df['Sentiment'].value_counts()

Positive    23998
Negative     8271
Neutral      5163
Name: Sentiment, dtype: int64

In [ ]:
ur_df[ur_df['Sentiment'] == 'Negative'].head()

App  ... Sentiment_Subjectivity
32  10 Best Foods for You  ...                  0.500
43  10 Best Foods for You  ...                  0.000
68  10 Best Foods for You  ...                  0.000
85  10 Best Foods for You  ...                  0.800
95  10 Best Foods for You  ...                  0.475

[5 rows x 5 columns]

In [ ]:
ur_df[ur_df['Sentiment'] == 'Neutral'].head()

App  ... Sentiment_Subjectivity
8   10 Best Foods for You  ...                    0.0
9   10 Best Foods for You  ...                    0.0
22  10 Best Foods for You  ...                    0.0
25  10 Best Foods for You  ...                    0.0
29  10 Best Foods for You  ...                    0.0

[5 rows x 5 columns]

In [ ]:
ur_df.tail(10)

App  ... Sentiment_Subjectivity
64285  Houzz Interior Design Ideas  ...                    NaN
64286  Houzz Interior Design Ideas  ...                    NaN
64287  Houzz Interior Design Ideas  ...                    NaN
64288  Houzz Interior Design Ideas  ...                    NaN
64289  Houzz Interior Design Ideas  ...                    NaN
64290  Houzz Interior Design Ideas  ...                    NaN
64291  Houzz Interior Design Ideas  ...                    NaN
64292  Houzz Interior Design Ideas  ...                    NaN
64293  Houzz Interior Design Ideas  ...                    NaN
64294  Houzz Interior Design Ideas  ...                    NaN

[10 rows x 5 columns]

In [ ]:
ur_df.shape

(64295, 5)

In [ ]:
ur_df.isnull().sum()

App                           0
Translated_Review         26868
Sentiment                 26863
Sentiment_Polarity        26863
Sentiment_Subjectivity    26863
dtype: int64